gcft_ui/rarc_tab.py

```diff
@@ -124,6 +124,100 @@ def extract_all_files_from_rarc_folder(self):
       file_type="RARC"
     )
   
+
+  def import_all_rarcs_from_folder(
+      self, 
+      folder_of_rarcs_path, 
+      recursive=True, 
+      base_path="", 
+      debug=True, rtn_counter=False, 
+      swallow_assertion_errors=True
+      ):
+
+    NO_RECURSION_DIRECTORIES = ["GCFTUnpacked","HomeBTN","\\res\\HomeBTN"] # FIles in HomeBTN throws some strange AssertionError
```

Comment:
What game is "HomeBTN" from? I want to see what the error is

Reply:
I was using the game Twilight Princess, NA Wii V1.0, built on 2006/10/10 20:35 (RVL-RZDE-0A-0). Also see https://www.zeldaspeedruns.com/tp/knowledge/version-differences.  The particular files are in files/res/homeBTN (homeBtn_ENG.arc, homeBtn_FRA.arc, and homeBtn_SPA.arc) and seem to be some type of language modifier (english vs french vs spanish?).
Here's a screenshot of the error with `swallow_assertion_errors` equal to `False`:
![RARC fail](https://github.com/user-attachments/assets/32ac1459-e8aa-4074-9235-5720abe03161)
They probably either aren't valid RARC files, or they are in some different format. I'll figure out more details tomorrow, but for now I'm going to bed.

Reply:
After further analysis, it seems that these files in particular are either invalid RARC files or (more likely) specifically formatted as something else. The "`magic`" (rarc.py line 43) is `Uｪ8-` (trips the assertion error, usually equal to `RARC`), which doesn't seem to be any format I know of. \
Opening the files using UTF-8 encoding in VS Code yields mostly gibberish with some intelligible fragments hidden within, such as `btryPwr_02_0` (battery power?) and `optnBtn_10_M` (option button?), giving the impression that the following data deals with each respective button's state (`HomeBTN` = Home Button?).

When opening the GCFT program normally and attempting to `Import RARC` with these files (using the UI's button), I get the following `UnicodeDecodeError` which further supports the "invalid RARC" theory.
![UnicodeErrorRegular](https://github.com/user-attachments/assets/68fa377d-155c-4b4d-9c5f-203db5ff6ee6)


I also added context to the printed error message in the console explaining that it's probably an invalid RARC file. Next, I'll add a printed warning in the console that lets you know if the `homeBTN` folder is or is within the folder you're trying to bulk decode and warns you that these are most likely invalid RARC files.

Reply:
There's no reason to hardcode the name of homebtn specifically. GCFT is a generic tool that should work with any gamecube game that uses the RARC format so it should have proper file format detection so that it skips an non-RARC files even if they have the .arc extension.

And the reverse is also true: it should detect if a file is a RARC even when its extension *isn't* .arc (many games do this).

Reply:
Understood, thanks. I updated the fork.